src/neo/Plugins/Plugin.cs

```diff
@@ -5,34 +5,35 @@
 using System.Linq;
 using System.Reflection;
 using System.Threading;
+using static System.IO.Path;
 
 namespace Neo.Plugins
 {
     public abstract class Plugin : IDisposable
     {
         public static readonly List<Plugin> Plugins = new List<Plugin>();
-        private static readonly List<ILogPlugin> Loggers = new List<ILogPlugin>();
+        internal static readonly List<ILogPlugin> Loggers = new List<ILogPlugin>();
         internal static readonly Dictionary<string, IStoragePlugin> Storages = new Dictionary<string, IStoragePlugin>();
         internal static readonly List<IRpcPlugin> RpcPlugins = new List<IRpcPlugin>();
         internal static readonly List<IPersistencePlugin> PersistencePlugins = new List<IPersistencePlugin>();
         internal static readonly List<IP2PPlugin> P2PPlugins = new List<IP2PPlugin>();
         internal static readonly List<IMemoryPoolTxObserverPlugin> TxObserverPlugins = new List<IMemoryPoolTxObserverPlugin>();
 
-        private static readonly string pluginsPath = Path.Combine(Path.GetDirectoryName(Assembly.GetEntryAssembly().Location), "Plugins");
+        public static readonly string PluginsDirectory = Combine(GetDirectoryName(Assembly.GetEntryAssembly().Location), "Plugins");
         private static readonly FileSystemWatcher configWatcher;
-
         private static int suspend = 0;
 
-        protected static NeoSystem System { get; private set; }
+        public virtual string ConfigFile => Combine(PluginsDirectory, GetType().Assembly.GetName().Name, "config.json");
         public virtual string Name => GetType().Name;
```

Comment:
Should we really make this virtual? Aren't we going to end in a similar situation described here https://github.com/neo-project/neo-cli/pull/452?

Reply:
https://github.com/neo-project/neo/pull/1312#issuecomment-560396819

Reply:
But Erik, if I understand it correctly, you can have a name for the plugin and a different name for the dll.  You are using it as default, but you are allowing people to change it (because of `virtual`).
I will test to see what happens.

Reply:
The answer is nothing happened. The `Name` is just a name.